lib/reverse_sentence.rb

```diff
@@ -1,6 +1,55 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  my_sentence = reverse_words(my_sentence)
+  return string_reverse(my_sentence)
+end
+
+def reverse_words(my_words)
+  return nil if my_words.nil?
+
+  start_word = 0
+  end_word = 0
+  i = 0
+  
+  unless my_words == nil
+    length = my_words.length
+    while i < length
+      while my_words[i] == " "
+        i += 1
+      end
+      start_word = i
+
+      while i < length && my_words[i] != " "
+        i += 1
+      end
+      end_word = i - 1
+
+      while end_word > start_word
```

Comment:
One way to clean up this fn would be to refactor your string_reverse function to reverse strings given a start and end index. Lines 29-34 could then be taken care of by the same fn.